core-api/src/main/java/com/optimizely/ab/event/internal/EventFactory.java

```diff
@@ -167,32 +167,34 @@ public LogEvent createConversionEvent(@Nonnull ProjectConfig projectConfig,
     private List<Attribute> buildAttributeList(ProjectConfig projectConfig, Map<String, ?> attributes) {
         List<Attribute> attributesList = new ArrayList<Attribute>();
 
-        for (Map.Entry<String, ?> entry : attributes.entrySet()) {
-            // Filter down to the types of values we're allowed to track.
-            // Don't allow Longs, BigIntegers, or BigDecimals - they /can/ theoretically be serialized as JSON numbers
-            // but may take on values that can't be faithfully parsed by the backend.
-            // https://developers.optimizely.com/x/events/api/#Attribute
-            if (entry.getValue() == null ||
-                    !((entry.getValue() instanceof String) ||
-                    (entry.getValue() instanceof Integer) ||
-                    (entry.getValue() instanceof Double) ||
-                    (entry.getValue() instanceof Boolean))) {
-                continue;
+        if (attributes != null) {
```

Comment:
Rather than a large enclosed if != how about if == null continue;  I think it reads better.

Reply:
Tom there is no for loop here and if attributes is not null then we have to build attribute list (for loop is inside if) from attributes and later on, we are also building attribute for bot filtering.